EIPS/eip-5131.md

```diff
@@ -0,0 +1,238 @@
+requires: 137
+--- 
+
+## Abstract
+Proving the ownership of an asset to a third party application in the Ethereum ecosystem is a common occurrence. Users frequently sign payloads of data presented to them via these applications in order to prove ownership. The process of proving ownership of an address is commonly used to authenticate users before allowing them to perform some operation.
+
+Examples:
+ - In order for you to edit your profile on OpenSea, you must sign a message with your wallet address.
+ - In order to access NFT gated content, you must sign a message with the wallet containing the NFT.
+ - In order to claim an airdrop, you must interact with the smart contract with the qualifying wallet address.
+ - In order to prove ownership of an NFT, you must sign a payload with the address that owns that NFT.
+
+Unfortunately, we've seen many cases where users have accidentally signed a malicious payload. The result is almost always a significant loss of assets associated with the signing address.
+
+In addition to this, many users keep significant portions of their assets in 'cold storage'. With the increased security from 'cold storage' solutions, we usually see decreased accessibility because users naturally increase the barriers required to access these wallets.
+
+This EIP proposes a solution which uses the Ethereum Name Service Specification (EIP-137) as a way to link one or more signing wallets (presumably less secure) to authenticate a main wallet.
+
+From there, the benefits are twofold. This EIP gives users increased security via outsourcing potentially malicious signing operations to wallets that are more accessible (hot wallets), while being able to maintain the intended security assumptions of wallets that are not frequently used for signing operations.
+
+
+## Motivation
+As explained in the abstract, the current method of using a person's main wallet for authentication of control and asset ownership is both insecure and inconvenient.  It is as if, in order to authenticate, you must give an application root access to your wallet.  This proposal provides an easy way to do 'read only' access to a wallet by leveraging existing infrastructure.
+
+Some solutions propose dedicated registry smart contracts to create this link, or new protocols to be supported.  Rather than 're-invent the wheel', this proposal aims to use the widely adopted Ethereum Name Service in order to bootstrap a safer and more convenient way to sign and authenticate.
+
+
+## Specification
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+
+Let:
+ - `mainAddress` represent the wallet address we are trying to authenticate or very asset ownership for
+ - `mainENS` represent the reverse lookup ENS string for `mainAddress`
+ - `authAddress` represent the address we want to sign with in lieu of `mainAddress`
```

Comment:
Are we going to add any requirement on this being an EOA?

Reply:
No I don't think so. On second thoughts, we shouldn't be restricting it because a smart contract address could be used pretty easily with the right setup.